### ISSUE TITLE
Expand Inline Suggest settings

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#49875]
- Problem statement

We get a lot of reports from different user groups that would like to have a better control over what suggestions are shown and where. While some like comment completions, some don't. Some do feel impacted when completions are shown immediately, and some don't like completions shown on specific operations in the editor.
Most of these are not solvable in the extension (where we listen to the inline suggest providers), as we do not get enough signals to handle these behaviors.

- Proposal

-- Settings:

- Disable on removal - Stops retrieval of new Inline Suggest completions while user is deleting. In many cases we observed that this would just produce the same line over again to the user, which is not always what users want.
- Disable on location (comment/string) - Same as we have for standard code completion (intellisense) - which would allow for a better control over where should completions render
- Show after *n* ms of idle - Would allow users to configure when suggestions are shown, to not break their workflows while coding, but e.g. show  suggestion only after they have stopped typing for 5 seconds and are stuck.

-- Nice to have:

It would be great to also add a signal to the Extension that a request has been blocked, delayed, dropped, so FE metrics could reason about why e.g. amount of provided suggestions was reduced.

-- Alternative:

Add additional signals to the context of the `provideInlineCompletion`, to allow extensions to make the call whether to add or not to add the filtering.
This would remove the need to add additional signals when the calls are dropped on the core end, and would allow extensions to stay in control of the inline completions as well as the logging/metrics.

Happy to help implement if there is no bandwidth and we agree on a way forward :)
